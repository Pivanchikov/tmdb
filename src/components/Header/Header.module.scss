@import '../../theme/colors.scss';

.header {
  position: sticky;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 64px;
  background-color: $darkBlue;

  &__content {
    max-width: 1300px;
    width: 100%;
    padding: 0 5%;
    display: flex;
    justify-content: space-between;
  }

  &__navigation {
    display: flex;
    align-items: center;

    img:first-child {
      margin-right: 16px;
    }
  }

  .toolbar {
    display: flex;
    align-items: center;
    justify-content: space-between;
    min-width: 200px;
  }

  .button {
    color: $white;
  }

  @media screen and (max-width: 960px) {
    .toolbar {
      min-width: auto;
    }
  }
}
