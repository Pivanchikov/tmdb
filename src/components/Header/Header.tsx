import React from 'react'
import { Hidden, Button } from '@material-ui/core'
import { useHistory } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

import main_logo from '../../assets/img/main_logo.svg'
import { LanguageSwitcher, BurgerMenu, MenuDropdown } from '../'
import classes from './Header.module.scss'
import { films, serials, people, another, burgerItems } from '../constants'
import { Search } from '../Search'

export const Header: React.FC = () => {
  const { t } = useTranslation()
  const history = useHistory()
  const redirectToMain = () => history.push('')

  return (
    <header className={classes.header}>
      <section className={classes.header__content}>
        <Hidden mdUp>
          <BurgerMenu items={burgerItems} />
        </Hidden>
        <div className={classes.header__navigation}>
          <img
            src={main_logo}
            alt="TMDB"
            width="154px"
            height="24px"
            data-testid="logo"
            onClick={redirectToMain}
          />
          <Hidden smDown>
            <MenuDropdown name="movies" items={films} />
            <MenuDropdown name="tvShows" items={serials} />
            <MenuDropdown name="people" items={people} />
            <MenuDropdown name="more" items={another} />
          </Hidden>
        </div>
        <Hidden mdDown>
          <Search />
        </Hidden>
        <div className={classes.toolbar}>
          <LanguageSwitcher />
          <Hidden smDown>
            <Button className={classes.button}>{t('signIn')}</Button>
            <Button className={classes.button}>{t('signUp')}</Button>
          </Hidden>
          <Hidden lgUp>
            <Search />
          </Hidden>
        </div>
      </section>
    </header>
  )
}
