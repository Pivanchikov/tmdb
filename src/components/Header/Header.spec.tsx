import { render as testingLibraryRender, fireEvent } from '@testing-library/react'

import { Header } from './Header'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}))

describe('Header', () => {
  const render = () => testingLibraryRender(<Header />)

  it('renders correctly logo', () => {
    const { getByTestId } = render()

    expect(getByTestId('logo')).toBeInTheDocument

    fireEvent.click(getByTestId('logo'))
  })

  it('matches snapshot', () => {
    const { asFragment } = render()

    expect(asFragment()).toMatchSnapshot()
  })
})
