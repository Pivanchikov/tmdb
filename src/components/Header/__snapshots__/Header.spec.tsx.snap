// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <section>
      <div>
        <img
          alt="TMDB"
          data-testid="logo"
          height="24px"
          src="SvgrURL"
          width="154px"
        />
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-languageSwitcher-13 MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label makeStyles-label-14"
          >
            en
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </section>
  </header>
</DocumentFragment>
`;
